Jenkinsfile

```diff
@@ -0,0 +1,247 @@
+                            // If this build is not a pull request, run full owasp scan. Otherwise run incremental scan
+                            script {
+                                if (params.RELEASE == true) {
+                                    sh "git checkout ${env.RELEASE_TAG}"
+                                }
+                                if (env.CHANGE_ID == null) {
+                                    sh 'mvn install -B -Powasp -DskipTests=true -nsu $DISABLE_DOWNLOAD_PROGRESS_OPTS'
+                                } else {
+                                    sh 'mvn install -B -Powasp -DskipTests=true  -Dgib.enabled=true -Dgib.referenceBranch=/refs/remotes/origin/$CHANGE_TARGET -nsu $DISABLE_DOWNLOAD_PROGRESS_OPTS'
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        stage('Quality Analysis') {
+            parallel {
+                // Sonar stage only runs against master
+                stage ('SonarCloud') {
+                    when {
+                        allOf {
+                            expression { env.CHANGE_ID == null }
+                            branch 'master'
+                            environment name: 'JENKINS_ENV', value: 'prod'
+                        }
+                    }
+                    steps {
+                        script {
+                            if (params.RELEASE == true) {
+                                sh "git checkout ${env.RELEASE_TAG}"
+                            }
+                        }
+                        withMaven(maven: 'M35', jdk: 'jdk11', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LINUX_MVN_RANDOM}') {
+                            withCredentials([string(credentialsId: 'SonarQubeGithubToken', variable: 'SONARQUBE_GITHUB_TOKEN'), string(credentialsId: 'cxbot-sonarcloud', variable: 'SONAR_TOKEN')]) {
+                                script {
+                                    sh 'mvn -q -B -Dcheckstyle.skip=true org.jacoco:jacoco-maven-plugin:prepare-agent install sonar:sonar -Dsonar.host.url=https://sonarcloud.io -Dsonar.login=$SONAR_TOKEN  -Dsonar.organization=cx -Dsonar.projectKey=${GITHUB_REPONAME} -Dsonar.exclusions=${COVERAGE_EXCLUSIONS} $DISABLE_DOWNLOAD_PROGRESS_OPTS'
+
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        stage('Release Tag') {
+            when { expression { params.RELEASE == true } }
+            steps {
+                script {
+                    sh "git checkout ${env.RELEASE_COMMIT}"
+
+                    //sshagent doesn't seem to work in multi-branch pipelines so the following hack is needed
+                    sh 'git remote add ssh-origin git@github.com:connexta/${env.GITHUB_REPONAME}.git'
```

Comment:
❗️  Use double quotes here instead of single quotes

Reply:
Should double quotes be used whenever we call the shell command in the Jenkinsfile?

Reply:
it depends, this is groovy and in groovy, there are many ways to do strings. In this case, the string is expecting `${env.GITHUB_REPONAME}` to be expanded and the only way to do that in groovy is with double quotes. Single quote are treated as is.